tests/tests/htlc_tests.cpp

```diff
@@ -177,6 +177,213 @@ try {
 } FC_LOG_AND_RETHROW()
 }
 
+class display_operation_history
+{
+   public:
+   display_operation_history( std::vector< operation_history_object > history, const account_object& acct );
+   const account_object& acct;
+};
+
+class op_printer
+{
+   public:
+   op_printer( std::ostream& out, display_operation_history* displayer, const operation_history_object& obj )
+         : out(out), caller(displayer), obj(obj) {}
+   typedef std::string result_type;
+
+   template<typename T>
+   std::string operator()(const T& op)const
+   {
+      return ""; 
+   }
+
+   std::string operator()(const htlc_redeemed_operation& op)const
+   {
+      out << "Called by " << caller->acct.name
+            << " Redeemer: " << id_to_string(op.redeemer) 
+            << " Preimage: " << preimage_to_string(op.preimage);
+      return "";
+   }
+
+   std::string operator()(const htlc_redeem_operation& op)const
+   {
+      out << "Called by " << caller->acct.name
+            << " Redeemer: " << id_to_string(op.redeemer) 
+            << " Preimage: " << preimage_to_string(op.preimage);
+      return "";
+   }
+
+   private:
+   std::ostream& out;
+   display_operation_history* caller;
+   const operation_history_object& obj;
+   static std::string id_to_string( object_id_type id) 
+         { return "" + std::to_string(id.space()) + "." + std::to_string(id.type()) + "." + std::to_string(id.instance()); }    
+   static std::string preimage_to_string( std::vector<char> in) { 
+         std::string out;
+         std::for_each(in.begin(), in.end(), [&out](const char& in) 
+               { out += std::to_string((int)in); });
+         return out; 
+      }
+};
+
+display_operation_history::display_operation_history( std::vector<operation_history_object> history, const account_object& acct)
+      : acct(acct)
+{
+   for( auto itr = history.begin(); itr != history.end(); itr++) 
+   {
+      std::stringstream ss;
+      (*itr).op.visit( op_printer(ss, this, *itr) );
+      if ( ss.rdbuf()->in_avail() > 0 )
+         BOOST_TEST_MESSAGE( ss.str() );
+   };
+}
+
+BOOST_AUTO_TEST_CASE( htlc_hf_bsip64 )
+{
+try {
+   ACTORS((alice)(bob)(joker));
+
+   int64_t init_balance(100 * GRAPHENE_BLOCKCHAIN_PRECISION);
+
+   transfer( committee_account, alice_id, graphene::chain::asset(init_balance) );
+   transfer( committee_account, joker_id, graphene::chain::asset(init_balance) );
+
+   advance_past_htlc_first_hardfork(this);
```

Comment:
We may need to cleanup code about the first hard fork at some point. Not necessarily in this release though.